lib/ironic_tls_setup.sh

```diff
@@ -23,74 +23,92 @@ if [ "${IRONIC_TLS_SETUP}" == "true" ]; then
     export MARIADB_CERT_FILE="${MARIADB_CERT_FILE:-"${WORKING_DIR}/certs/mariadb.crt"}"
     export MARIADB_KEY_FILE="${MARIADB_KEY_FILE:-"${WORKING_DIR}/certs/mariadb.key"}"
 
+    # BMO run-local scripts automatically enables iPXE TLS if the certs are
+    # present (BMO tooling does this with all certs) and iPXE TLS require
+    # custom firmware building in dev-env thus this condition syncs
+    # the cert generation with the firmware building
+    export IPXE_ENABLE_TLS="${IPXE_ENABLE_TLS:-false}"
+    export BUILD_IPXE="${BUILD_IPXE:-false}"
     export IPXE_CACERT_FILE="${IPXE_CACERT_FILE:-"${WORKING_DIR}/certs/ipxe-ca.pem"}"
     export IPXE_CAKEY_FILE="${IPXE_CAKEY_FILE:-"${WORKING_DIR}/certs/ipxe-ca.key"}"
     export IPXE_CERT_FILE="${IPXE_CERT_FILE:-"${WORKING_DIR}/certs/ipxe.crt"}"
     export IPXE_KEY_FILE="${IPXE_KEY_FILE:-"${WORKING_DIR}/certs/ipxe.key"}"
 
     # Generate CA Key files
-    if [ ! -f "${IRONIC_CAKEY_FILE}" ]; then
+    if [[ ! -r "${IRONIC_CAKEY_FILE}" ]]; then
         openssl genrsa -out "${IRONIC_CAKEY_FILE}" 2048
     fi
-    if [ ! -f "${IRONIC_INSPECTOR_CAKEY_FILE}" ]; then
+    if [[ ! -r "${IRONIC_INSPECTOR_CAKEY_FILE}" ]]; then
         openssl genrsa -out "${IRONIC_INSPECTOR_CAKEY_FILE}" 2048
     fi
-    if [ ! -f "${MARIADB_CAKEY_FILE}" ]; then
+    if [[ ! -r "${MARIADB_CAKEY_FILE}" ]]; then
         openssl genrsa -out "${MARIADB_CAKEY_FILE}" 2048
     fi
-    if [ ! -f "${IPXE_CAKEY_FILE}" ]; then
+    if [[ ! -r "${IPXE_CAKEY_FILE}" ]]  && \
+       [[ "${IPXE_ENABLE_TLS,,}" == "true" ]] && \
```

Comment:
This test with `IPXE_ENABLE_TLS` and `BUILD_IPXE` are repeated quite a few times . Maybe have the test result stored in intermediate variable to improve maintainability.?

Reply:
Thanks, done.